src/main/java/budgetbuddy/logic/parser/commandparsers/rulecommandparsers/RuleSwapCommandParser.java

```diff
@@ -0,0 +1,37 @@
+package budgetbuddy.logic.parser.commandparsers.rulecommandparsers;
+
+import static budgetbuddy.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import budgetbuddy.commons.core.index.Index;
+import budgetbuddy.logic.commands.rulecommands.RuleSwapCommand;
+import budgetbuddy.logic.parser.CommandParser;
+import budgetbuddy.logic.parser.CommandParserUtil;
+import budgetbuddy.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new RuleSwapCommand object.
+ */
+public class RuleSwapCommandParser implements CommandParser<RuleSwapCommand> {
+    @Override
+    public String name() {
+        return RuleSwapCommand.COMMAND_WORD;
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RuleSwapCommand
+     * and returns an RuleSwapCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public RuleSwapCommand parse(String args) throws ParseException {
+        String[] indexPair = args.trim().split("\\s+", 2);
```

Comment:
Might want to get rid of the limit, so it's just `args.trim().split("\\s+")`. If the user enters `rule swap 1 2 3` and the swap works with rules 1 and 2 while 3 is ignored, it might be reported as a bug due to a lack of error message.

Reply:
Fair point!